archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/ArchUnitExtensionContext.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.junit;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestInstances;
+import org.junit.platform.engine.TestTag;
+import org.junit.platform.engine.reporting.ReportEntry;
+
+import static java.util.stream.Collectors.toSet;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+class ArchUnitExtensionContext implements ExtensionContext {
+
+    private final AbstractArchUnitTestDescriptor descriptor;
+    private final ArchUnitEngineExecutionContext context;
+
+    public ArchUnitExtensionContext(
+            AbstractArchUnitTestDescriptor descriptor,
+            ArchUnitEngineExecutionContext execution) {
+        this.descriptor = descriptor;
+        this.context = execution;
+    }
+
+    @Override
+    public Optional<ExtensionContext> getParent() {
```

Comment:
Not sure exactly which methods are used by JUnit conditions, preferred to give each of the method a sensible implementation, though